### PR TITLE
Migrate updateExperiment endpoint.

### PR DESCRIPTION
**Pull request by dsuhinin**
PR related to -> https://github.com/users/jgiannuzzi/projects/2/views/1?pane=issue&itemId=72871946 task.

### CODE REVIEW

File: pkg/tracking/store/sql/experiments.go [hunk 1]
```diff
@@ -78,6 +78,17 @@ func (s TrackingSQLStore) CreateExperiment(input *protos.CreateExperiment) (stri
 	return strconv.Itoa(int(*experiment.ID)), nil
 }
 
+func (s TrackingSQLStore) UpdateExperiment(experiment *protos.Experiment) *contract.Error {
```
Comment (nojaf) [inline pkg/tracking/store/sql/experiments.go:81]: @jgiannuzzi the Python code went with `rename_experiment`. Do we want to do the same here?

Comment (jgiannuzzi) [reply pkg/tracking/store/sql/experiments.go:81]: yes, I think we should keep the Go code _naming_ identical to the Python code - if we strongly feel that the name is badly chosen or inconsistent, we should probably raise a PR upstream but keep in sync in the meantime

File: pkg/tracking/service/experiments.go [hunk 2]
```diff
@@ -89,6 +90,23 @@ func (ts TrackingService) RestoreExperiment(
...
 
+func (ts TrackingService) UpdateExperiment(
+	input *protos.UpdateExperiment,
+) (*protos.UpdateExperiment_Response, *contract.Error) {
+	experiment, err := ts.Store.GetExperiment(input.GetExperimentId())
+	if err != nil {
+		return nil, err
+	}
+	if input.NewName != nil {
+		experiment.Name = utils.PtrTo(input.GetNewName())
```
Comment (nojaf) [inline pkg/tracking/service/experiments.go:101]: Not feeling strongly about this one but would it not make sense to just assign `experiment.Name = input.NewName` I would assume these are both pointers. These are the things where I'm not sure what the proper Go thing to do it.

File: pkg/tracking/store/sql/experiments.go [hunk 3]
```diff
@@ -78,6 +78,17 @@ func (s TrackingSQLStore) CreateExperiment(input *protos.CreateExperiment) (stri
 	return strconv.Itoa(int(*experiment.ID)), nil
 }
 
+func (s TrackingSQLStore) RenameExperiment(experiment *protos.Experiment) *contract.Error {
+	if err := s.db.Model(&models.Experiment{}).
```
Comment (nojaf) [inline pkg/tracking/store/sql/experiments.go:82]: I don't think we are checking if the experiment life cycle is Active.
Python code does:
```
            if experiment.lifecycle_stage != LifecycleStage.ACTIVE:
                raise MlflowException("Cannot rename a non-active experiment.", INVALID_STATE)
```
We really want to do the same thing there.

**Review by nojaf — CHANGES_REQUESTED** [review]
Looking good, I think we want to respect the lifecycle check.

File: pkg/tracking/service/experiments.go (hunk 2 above)
Comment (nojaf) [reply pkg/tracking/service/experiments.go:108]: I don't mind the `if` check, but just wonder whether `input.GetNewName()` (getting string value from pointer) only to wrap it again as a pointer is what we want here. It feels like an extra step.

**Review by nojaf — APPROVED** [review]

**Review by nojaf — APPROVED** [review]

**Pull request merged**